src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -619,58 +619,96 @@ private int decodePublish(
             {
                 reasonCode = PROTOCOL_ERROR;
             }
-
-            if (reasonCode == 0)
+            else
             {
-                final int topicKey = topicKey(topicName);
-                MqttServer.MqttServerStream publisher = server.streams.get(topicKey);
-
-                if (publisher == null)
+                boolean hasTopicAlias = false;
+                decodeProperties:
+                for (int decodeProgress = decodeOffset; decodeProgress < decodeLimit; )
                 {
-                    publisher = server.resolvePublisher(traceId, authorization, topicName);
-                    if (publisher == null)
+                    final MqttPropertyFW mqttProperty = mqttPropertyRO.wrap(decodeBuffer, decodeProgress, decodeLimit);
+                    switch (mqttProperty.kind())
                     {
-                        server.decodePublisherKey = 0;
-                        server.decodeablePacketBytes = 0;
-                        server.decoder = decodePacketType;
-                        progress = publish.limit();
-                        break decode;
+                    case KIND_EXPIRY_INTERVAL:
+                        expiryInterval = mqttProperty.expiryInterval();
+                        break;
+                    case KIND_CONTENT_TYPE:
+                        contentType = mqttProperty.contentType().asString();
+                        break;
+                    case KIND_TOPIC_ALIAS:
+                        if (hasTopicAlias)
+                        {
+                            reasonCode = PROTOCOL_ERROR;
+                            break decodeProperties;
+                        }
+
+                        hasTopicAlias = true;
+                        int alias = mqttProperty.topicAlias() & 0xFFFF;
+
+                        if (alias <= 0 || alias > server.topicAliasMaximum)
+                        {
+                            reasonCode = TOPIC_ALIAS_INVALID;
+                            break decodeProperties;
+                        }
+
+                        if (topic.isEmpty())
+                        {
+                            if (!server.topicAliases.containsKey(alias))
+                            {
+                                reasonCode = PROTOCOL_ERROR;
+                                break decodeProperties;
+                            }
+                            topic = server.topicAliases.get(alias);
+                        }
+                        else
+                        {
+                            server.topicAliases.put(alias, topic);
+                        }
+                        break;
+                    case KIND_PAYLOAD_FORMAT:
+                        payloadFormat = MqttPayloadFormat.valueOf(mqttProperty.payloadFormat());
+                        break;
+                    case KIND_RESPONSE_TOPIC:
+                        responseTopic = mqttProperty.responseTopic().asString();
```

Comment:
Given that the underlying bytes are stable and remain on stack, can we avoid the `toString()` calls by wrapping the underlying bytes using a `String16FW responseTopicRO` on the factory, and passing that instead of `String` to `onDecodePublishSuccess`.